File: laocai-bot-spring-boot-starter/src/main/java/indi/dkx/laocai/bot/model/segment/MentionSegment.java
package indi.dkx.laocai.bot.model.segment;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.Objects;

/**
 * @提及消息段。
 */
public final class MentionSegment implements Segment {

    record Data(Long user_id) {
        MentionSegment toSegment() {
            return new MentionSegment(this);
        }
    }

    private final Data data;

    private MentionSegment(Data data) {
        this.data = data;
    }

    public static MentionSegment of(Long userId) {
        return new MentionSegment(new Data(Objects.requireNonNull(userId)));
    }

    @Override
    public String getType() {
        return "mention";
    }

    @Override
    public Data getData() {
        return data;
    }

    @JsonIgnore
    public Long getUserId() {
        return data.user_id();
    }
}

File: laocai-bot-spring-boot-starter/src/main/kotlin/indi/kyson/laocai/bot/segment/MentionSegment.kt
package indi.kyson.laocai.bot.segment

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @提及消息段。
 */
class MentionSegment private constructor(override val data: Data) : Segment {

    data class Data(@JsonProperty("user_id") val userId: Long) {
        fun toSegment(): MentionSegment = MentionSegment(this)
    }

    override val type: String
        get() = "mention"

    @get:JsonIgnore
    val userId: Long
        get() = data.userId

    companion object {
        @JvmStatic
        fun of(userId: Long): MentionSegment = MentionSegment(Data(userId))
    }
}
